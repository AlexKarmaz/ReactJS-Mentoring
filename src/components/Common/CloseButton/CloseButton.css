.closeButton {
    padding: 0;
    margin: 0;
    cursor: pointer;
    outline: none;
    justify-content: center;
    display: flex;
    align-items: center;
    background-color: transparent;
    border: #232323;
}

.closeButton.medium{
    width: 20px;
    height: 20px;
}

.closeButton.small{
    width: 15px;
    height: 15px;
}

.closeButton.medium > img {
    height: 20px;
    width: 20px;
}

.closeButton.small > img {
    height: 15px;
    width: 15px;
}
