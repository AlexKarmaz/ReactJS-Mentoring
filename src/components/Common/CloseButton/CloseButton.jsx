import React from 'react';
import './CloseButton.css'
import closeIcon from '../../../../public/images/icon-x-white.png';

//PATTERN: Stateless component
const CloseButton = ({size='medium', onClick}) => (
    <button className={`closeButton ${size}`} onClick={onClick}>
        <img className='closeButton-image' src={closeIcon} />
    </button>
);

export default CloseButton;
