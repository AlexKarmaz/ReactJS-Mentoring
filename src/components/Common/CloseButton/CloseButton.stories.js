/* eslint-disable*/
import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import CloseButton from './CloseButton';

const wrapperStyle = {
  backgroundColor: '#232323',
};

export default {
  title: 'Close Button',
  component: CloseButton,
  decorators: [withKnobs],
};

export const Default = () => (
  <div  style={wrapperStyle}>
    <CloseButton
    onClick={action('CloseButton-click')}
  />
  </div>
);
