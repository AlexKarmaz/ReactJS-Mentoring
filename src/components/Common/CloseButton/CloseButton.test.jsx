import { shallow } from 'enzyme';
import React from 'react';
import CloseButton from './CloseButton';

describe('CloseButton', () => {
    it('should be rendered', () => {
        const onClose = jest.fn();

        const closeButton = shallow(<CloseButton onClick={onClose} size='small' />);
        closeButton.simulate('click');

        expect(closeButton.find('button').hasClass('closeButton small')).toBeTruthy();
        expect(onClose).toBeCalled();
        expect(closeButton).toMatchSnapshot();
    });
});
