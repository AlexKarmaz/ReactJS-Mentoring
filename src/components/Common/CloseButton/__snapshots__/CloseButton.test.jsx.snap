// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton should be rendered 1`] = `ShallowWrapper {}`;
